nixos/modules/services/networking/bind.nix

```diff
@@ -153,18 +161,19 @@ in
 
       zones = mkOption {
         default = [];
-        type = types.listOf (types.submodule [ { options = bindZoneOptions; } ]);
+        type = with types; coercedTo (listOf attrs) bindZoneCoerce (attrsOf (types.submodule bindZoneOptions));
```

Comment:
Shouldn't backwards compatibility be considered for existing configs? Or is there a reason you can't use `types.oneOf [list attrset]`?

Reply:
As far as I understood and tested, this is where `coercedTo` comes into play.
Both the old syntax and the new syntax should work fine, and is also what I got from testing it :)

Reply:
👍

Reply:
This is also covered in a bind9 test :)